service/lib/dinstaller/storage/manager.rb

```diff
@@ -58,7 +58,7 @@ def install(_progress)
       #
       # @return [Storage::Proposal]
       def proposal
-        @proposal ||= Proposal.new(logger, config)
+        @proposal ||= Proposal.new(logger, config, actions)
```

Comment:
Maybe we can use the callbacks approach instead of passing objects:

~~~
@proposal ||= Proposal.new(logger, config).tap do |proposal|
  proposal.on_change { actions.changed! }
end
~~~

Reply:
Well, it depends who is the resposible of orchestrating what happens when a proposal (or whatever object) changes. For example, if we consiser that `Manager` should do that work, then I would also do something like:

~~~
@software ||= Software.new(logger, config).tap do |software|
  software.on_product_change { self.probe } 
end
~~~

But, if we consider that `Proposal` should know when to call `Actions` callbacks, then I would keep as it is.

Reply:
For me it is more about who depends on what. If actions depends on proposal or other way. So for me nature way is to pass proposal to actions ( or use callback when creating actions ) as actions depends on proposal and not vice versa.

Regarding that if we should use for depedencies callbacks or passing directly object. It does not matter too much for me, only if there is more uses for callbacks, then callbacks is easier to share.

Reply:
In general, I prefer the callback approach for the same reason that @joseivanlopez pointed out: decoupling pieces. For instance, the software layer *does not depend on the manager* to do its work. Actually, now that we have moved the software logic to a separate service, I would expect to just raise a signal and forget about the rest.

Of course, I know that this PR is not based on the *multi-process* branch but on *master*. So unless you plan to keep working on this, I would like to adapt these changes and continue working from there on top of *multi-process* changes.